.github/workflows/ci.yml

```
@@ -90,17 +90,36 @@ jobs:
     runs-on: ubuntu-latest
     needs: [lint, check-msrv, examples]
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - uses: messense/maturin-action@v1
         with:
           target: aarch64
           manylinux: auto
           args: --manifest-path examples/simple/Cargo.toml
 
+  address-sanitizer:
+    runs-on: ubuntu-22.04
+    needs: [lint, check-msrv, examples]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          profile: minimal
+          default: true
+      - uses: Swatinem/rust-cache@v2
+        continue-on-error: true
+      - run: |
+            pip install numpy
+            cargo test --target x86_64-unknown-linux-gnu --release --lib --tests
```

You might also want to use `-Zbuild-std`

```suggestion
            cargo test -Zbuild-std --target x86_64-unknown-linux-gnu --release --lib --tests
```

https://doc.rust-lang.org/beta/unstable-book/compiler-flags/sanitizer.html#instrumentation-of-external-dependencies-and-std

---

Good point. Thanks!